IntegrationTests/IntegrationScoped.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace IntegrationTests
+{
+    public class IntegrationScoped<TStrategy> where TStrategy : IScopedStrategy, new()
+    {
+
+        protected readonly TStrategy Strategy = new TStrategy();
+
+        [TestInitialize]
+        public void Setup()
+        {
+            Strategy.Setup().Wait();
```

Comment:
@RhysC You have to wait unfortunately setup does not allow task as results in mstest :( You can still parallelise the work in individual setups so thats better than nothing.